core/iwasm/include/wasm_export.h

```diff
@@ -716,6 +716,15 @@ wasm_runtime_thread_env_inited(void);
 WASM_RUNTIME_API_EXTERN wasm_module_inst_t
 wasm_runtime_get_module_inst(wasm_exec_env_t exec_env);
 
+/**
+ * Get suspend flags from the execution environment
+ * 
+ * @param exec_env the execution environment to get the suspend flags from
+ * @return the suspend flags 
+ */
+WASM_RUNTIME_API_EXTERN uint32_t
```

Comment:
Is it a little heavy to exposing the API?
How about adding API in `core/iwasm/libraries/thread-mgr/thread_manager.h/.c`:

```C
bool
wasm_cluster_is_thread_terminated(WASMExecEnv *exec_env)
{
    return (exec_env->suspend_flags.flags & 0x01) ? true : false;
}
```
And call it in libc_wasi_wrapper.c:
```C
#if WASM_ENABLE_THREAD_MGR != 0
#include "../../../thread-mgr/thread_manager.h"
#endif
```